signalfx-java/src/main/java/com/signalfx/signalflow/ComputationHandler.java

```
@@ -0,0 +1,147 @@
+    }
+    
+    @Override
+    public Computation call() throws ComputationAbortedException, ComputationFailedException,
+            SignalFlowException, IllegalStateException {
+
+        if (this.computation.getState() == State.STATE_COMPLETED) {
+            throw new IllegalStateException("computation is completed");
+        }
+        
+        this.startTimeMs = System.currentTimeMillis();
+        this.stopTimeMs = null;
+        
+        try {
+            // iterate computation messages and route to message handling methods
+            for (ChannelMessage message : computation) {
+
+                switch (message.getType()) {
+
+                case JOB_START:
+                    JobStartMessage jobStartMessage = (JobStartMessage) message;
+                    onMessage(jobStartMessage);
+                    break;
+
+                case JOB_PROGRESS:
+                    JobProgressMessage jobProgressMessage = (JobProgressMessage) message;
+                    onMessage(jobProgressMessage);
+                    break;
+
+                case DATA_MESSAGE:
+                    DataMessage dataMessage = (DataMessage) message;
+                    onMessage(dataMessage);
+                    break;
+
+                case EVENT_MESSAGE:
+                    EventMessage eventMessage = (EventMessage) message;
+                    onMessage(eventMessage);
+                    break;
+
+                case METADATA_MESSAGE:
+                    MetadataMessage metadataMessage = (MetadataMessage) message;
+                    onMessage(metadataMessage);
+                    break;
+
+                default:
+                    break;
+                }
+            }
+        } finally {
+            this.stopTimeMs = System.currentTimeMillis();
+            close();
+        }
+
+        return this.computation;
+    }
+
+    @Override
+    public void run() {
+        try {
+            call();
+        } catch (ComputationAbortedException ex) {
```

If you're providing this class as a base for people to build on, you shouldn't impose on them that all errors are ignored and logged.